src/main/kotlin/org/jetbrains/research/testspark/appstarter/TestSparkStarter.kt

```diff
@@ -0,0 +1,219 @@
+        val classPath = "$projectPath${ToolUtils.sep}${args[4]}"
+        // JUnit Version
+        val jUnitVersion = args[5]
+        // Selected mode
+        val model = args[6]
+        // Token
+        val token = args[7]
+        // A txt file containing the prompt template
+        val promptTemplateFile = args[8]
+        // Output directory
+        val output = args[9]
+
+        println("Test generation requested for $projectPath")
+
+        ApplicationManager.getApplication().invokeAndWait {
+            val project = ProjectUtil.openOrImport(projectPath, null, true) ?: run {
+                println("couldn't find project in $projectPath")
+                exitProcess(1)
+            }
+            println("Detected project: $project")
+            // Continue when the project is indexed
+            println("Indexing project...")
+            project.let {
+                DumbService.getInstance(it).runWhenSmart {
+                    // open target file
+                    val virtualFile = LocalFileSystem.getInstance().findFileByPath(filePath) ?: run {
+                        println("couldn't open file $filePath")
+                        exitProcess(1)
+                    }
+
+                    // get target PsiClass
+                    val psiFile = PsiManager.getInstance(project).findFile(virtualFile) as PsiJavaFile
+                    val targetPsiClass = detectPsiClass(psiFile.classes, classUnderTestName) ?: run {
+                        println("couldn't find $classUnderTestName in $filePath")
+                        exitProcess(1)
+                    }
+
+                    println("PsiClass ${targetPsiClass.qualifiedName} is detected! Start the test generation process.")
+
+                    // update settings
+                    val settingsState = project.getService(LLMSettingsService::class.java).state
+                    settingsState.currentLLMPlatformName = LLMDefaultsBundle.get("grazieName")
+                    settingsState.grazieToken = token
+                    settingsState.grazieModel = model
+                    settingsState.classPrompts = JsonEncoding.encode(mutableListOf(File(promptTemplateFile).readText()))
+                    settingsState.junitVersion = when (jUnitVersion.filter { it.isDigit() }) {
+                        "4" -> JUnitVersion.JUnit4
+                        "5" -> JUnitVersion.JUnit5
+                        else -> {
+                            throw IllegalArgumentException("JUnit version $jUnitVersion is not supported. Supported JUnit versions are '4' and '5'")
+                        }
+                    }
+                    project.service<PluginSettingsService>().state.buildPath = classPath
+
+                    // Prepare Project Context
+                    // First, get CUT Module
+                    val cutModule =
+                        ProjectFileIndex.getInstance(project)
```

Comment:
What is CUT module?

Reply:
The module in which the class under test exists. Each project open with IJ can have various modules.